packages/client/src/views/Settings/items/Language.tsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import * as React from 'react'
+import {
+  ListViewItemSimplified,
+  ResponsiveModal,
+  FloatingNotification,
+  NOTIFICATION_TYPE
+} from '@opencrvs/components/lib/interface'
+import { useIntl, FormattedMessage } from 'react-intl'
+import {
+  userMessages,
+  buttonMessages,
+  constantsMessages
+} from '@client/i18n/messages'
+import {
+  LabelContainer,
+  ValueContainer,
+  DynamicHeightLinkButton,
+  Message,
+  Label
+} from '@client/views/Settings/items/utils'
+import { useSelector, useDispatch } from 'react-redux'
+import { IStoreState } from '@client/store'
+import { ILanguageState } from '@client/i18n/reducer'
+import { TertiaryButton, PrimaryButton } from '@opencrvs/components/lib/buttons'
+import { Select } from '@opencrvs/components/lib/forms'
+import { getAvailableLanguages } from '@client/i18n/utils'
+import { IUserDetails } from '@client/utils/userUtils'
+import { getUserDetails } from '@client/profile/profileSelectors'
+import { changeLanguage as changeLanguageActionCreator } from '@client/i18n/actions'
+
+interface ILanguageOptions {
+  [key: string]: string
+}
+
+export function Language() {
+  const [showLanguageSettings, setShowLanguageSettings] =
+    React.useState<boolean>(false)
+  const [showSuccessNotification, setShowSuccessNotification] =
+    React.useState<boolean>(false)
+  const intl = useIntl()
+  const languages = useSelector<IStoreState, ILanguageState>(
+    (state) => state.i18n.languages
+  )
+  const language = useSelector<IStoreState, string>(
+    (state) => state.i18n.language
+  )
+  const [selectedLanguage, setSelectedLanguage] =
+    React.useState<string>(language)
+  const toggleLanguageSettingsModal = React.useCallback(() => {
+    setShowLanguageSettings((prevValue) => !prevValue)
+  }, [])
```

Comment:
Thanks @Zangetsu101, I thought that it was given as a prop for the [change button ](https://github.com/opencrvs/opencrvs-core/blob/e163293547088c653632b499a4abdb81d8c119a8/packages/client/src/views/Settings/items/Language.tsx#L118) and to prevent changing the function value

Reply:
Even if we prevent changing the function value, it's still gonna re-render right? Because I don't think we've used `React.memo` to memoize the props or have we?

Reply:
Yes, it would re-render because I didn't memoize the button component. Thanks, I am going to remove `React.useCallback` here

Reply:
The required refactor has been done in https://github.com/opencrvs/opencrvs-core/pull/3647/commits/078caa93994ac77420f6cc14c4cc80dd424bb720